addons/controllers/certificates_test.go

```diff
@@ -0,0 +1,142 @@
+			certFilePath := path.Join(webhookServer.CertDir, webhookServer.CertName)
+			Expect(certFilePath).ShouldNot(BeEmpty())
+			keyFilePath := path.Join(webhookServer.CertDir, webhookServer.KeyName)
+			Expect(keyFilePath).ShouldNot(BeEmpty())
+			Expect(webhookServer).ToNot(BeNil())
+			secret, err := certificates.UpdateTLSSecret(ctx, webhookScrtName, webhookServiceName, webhookServer, addonNamespace)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(secret).NotTo(BeNil())
+			cert, err := cert2.CertsFromFile(certFilePath)
+			Expect(err).ToNot(HaveOccurred())
+			certPEM, err := cert2.EncodeCertificates(cert[0])
+			Expect(err).ToNot(HaveOccurred())
+			Expect(certPEM).To(Equal(secret.Data[resources.ServerCert]))
+			key, err := keyutil.PrivateKeyFromFile(keyFilePath)
+			Expect(err).ToNot(HaveOccurred())
+			orgKey, err := keyutil.ParsePrivateKeyPEM(secret.Data[resources.ServerKey])
+			Expect(err).ToNot(HaveOccurred())
+			Expect(key).To(Equal(orgKey))
+		})
+		It("should become invalid after one week", func() {
+			secret, err := certificates.UpdateTLSSecret(ctx, webhookScrtName, webhookServiceName, webhookServer, addonNamespace)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(secret).NotTo(BeNil())
+			err = certificates.ValidateCertificate(secret, time.Hour*24) // valid cert life is one week. One day should not make it invalid
+			Expect(err).ShouldNot(HaveOccurred())
+			err = certificates.ValidateCertificate(secret, 8*time.Hour*24) // in 8 days certificate should be  invalid
+			Expect(err).Should(HaveOccurred())
+
+		})
+	})
+
+	Context("Mutating and Validating Configurations", func() {
+		It("should be updated with CA bundle ", func() {
+			var err error
+
+			se := adminregv1.SideEffectClassNone
+			srvc := adminregv1.ServiceReference{
+				Namespace: addonNamespace,
+				Name:      webhookServiceName,
+				Path:      nil,
+				Port:      nil,
+			}
+
+			validatingWebhookConfig := &adminregv1.ValidatingWebhookConfiguration{}
+			validatingWebhookConfig.Name = "validating-webhook-config"
+			validatingWebhookConfig.Webhooks = []adminregv1.ValidatingWebhook{
+				{
+					Name: "vcalicoconfig.kb.io",
+					ClientConfig: adminregv1.WebhookClientConfig{
+						Service:  &srvc,
+						CABundle: []byte{'b', 'a', 'd'}},
+					SideEffects:             &se,
+					AdmissionReviewVersions: []string{"v1"},
+				},
+			}
+			Expect(k8sClient.Create(ctx, validatingWebhookConfig)).To(Succeed())
+
+			defaultingWebhookConfig := &adminregv1.MutatingWebhookConfiguration{}
+			defaultingWebhookConfig.Name = "mutating-webhook-config"
+			defaultingWebhookConfig.Webhooks = []adminregv1.MutatingWebhook{
+				{
+					Name: "mcalicoconfig.kb.io",
+					ClientConfig: adminregv1.WebhookClientConfig{
+						Service:  &srvc,
+						CABundle: []byte{'b', 'a', 'd'}},
+					SideEffects:             &se,
+					AdmissionReviewVersions: []string{"v1"},
+				},
+			}
```

Comment:
This whole section creating webhook configurations should be done through testuilts.CreateResoruces(), that way later on we can just point it to the yaml files we will use to deploy the real pacakge.

Reply:
Lets leave it here for now.